components/remote_settings/src/error.rs

```diff
@@ -47,6 +47,8 @@ pub enum Error {
     ConfigError(String),
     #[error("Database error: {0}")]
     DatabaseError(#[from] rusqlite::Error),
+    #[error("No attachement in given record: {0}")]
```

Comment:
This error covers a bit more than that. Should we have a distinct error like `RecordAttachmentMismatchError` for when downloaded data doesn't match the server?